src/float.js

```diff
@@ -884,11 +884,13 @@ Sk.builtin.float_.prototype.str$ = function (base, sign) {
 
     if (base === undefined || base === 10) {
         if (Sk.__future__.python3) {
-            tmp = work.toPrecision(16);
+            // 17 would get another number string
+            var tmpCheckString = work.toString();
+            var tmpCheckPrecision = work.toPrecision(17);
+            tmp = tmpCheckString === tmpCheckPrecision?  tmpCheckPrecision: work.toPrecision(16);
```

Comment:
can you clean up the white space here? thanks

Reply:
OK

Reply:
Now is ok？